jax/experimental/mosaic/gpu/launch_context.py

```diff
@@ -226,6 +226,21 @@ def batch(self, leading_rank: int) -> MemRefTransform:
     raise NotImplementedError  # Unused
 
 
+@dataclasses.dataclass(frozen=True)
+class PeerMemTransform(MemRefTransform):
```

Comment:
It doesn't really make sense to keep this as a Mosaic GPU transform to be honest. The only reason why we use those is for TMA. It does make sense to keep it as a Pallas transform, and you can move this implementation to there if needed